tlatools/org.lamport.tlatools/src/util/SimpleFilenameToStream.java

```diff
@@ -136,112 +118,94 @@ private String[] getLibraryPaths(final String installationBasePath, String[] lib
       }
       path = buf.toString();
     }
-    if (path == null) {
-      res = new String[1];
-      res[0] = installationBasePath + FileUtil.separator;
-    }
-    else {
+    if (path != null) {
       String[] paths = path.split(FileUtil.pathSeparator);
-      res = new String[paths.length+1];
-      for(int i=0;i<paths.length;i++){
-	res[i] = paths[i];
-	if (!res[i].endsWith(FileUtil.separator)){
-	  res[i] = res[i] + FileUtil.separator;
-	}
+      for (String s : paths) {
+          res.add(new FilesystemResolver(Path.of(s)));
       }
-      res[paths.length] = installationBasePath + FileUtil.separator;
     }
-    return res;
+
+    // Finally, search the bare classpath (e.g. for the community modules)
+    res.add(new ClasspathResolver(cl, ""));
+
+    return new SequentialResolver(res);
```

Comment:
Since this line will fundamentally be what we look at in the future when figuring out the resolver search order, I think using the `Resolver...` API suggested above would make it very clear.

Reply:
I won't add a `...` constructor (see notes on the other comment).

BUT, I am very sympathetic to "this particular block needs to be as clear as possible". What I will do is promise a follow-up PR to improve the clarity of this section. The final code might look something like
```java
return new SequentialResolver(List.of(
    searchUserDirOrCWD(),
    searchFilesystemPaths(libraries),
    searchBareClasspath()));
```

(This block of code is in need of other changes as well which are best addressed in a separate PR. For instance, it constructs a `":"`-separated String, only to immediately destruct that String to get the constituent paths.)